lib/src/screens/setup_2fa/widgets/popup_cancellable_alert.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:cake_wallet/src/widgets/alert_close_button.dart';
+import 'package:cake_wallet/src/widgets/primary_button.dart';
+import 'package:flutter/material.dart';
+import 'package:cake_wallet/palette.dart';
+
+import 'package:cake_wallet/src/widgets/alert_background.dart';
+
+class PopUpCancellableAlertDialog extends StatelessWidget {
+  final String contentText;
+  final String actionButtonText;
+  final VoidCallback? buttonAction;
+  final bool sameActionForButtonAndClose;
+
+  const PopUpCancellableAlertDialog({
+    super.key,
+    this.contentText = '',
+    this.actionButtonText = '',
+    this.buttonAction,
+    this.sameActionForButtonAndClose = true,
```

Comment:
It is to handle the scenario where the user clicks the cancel button instead of the 'Okay' button.

If this happens, this check acts as a failsafe for it and runs the same function as if the main 'Okay' button is clicked